Rewrite the app in Rust without rebasing the original boilerplate commits

davidyuk opened the pull request:
implements the same as #24

CedrikNikita commented on src/signTransaction.c:
```diff
@@ -36,33 +43,29 @@ static unsigned int ui_approval_nanos_button(unsigned int button_mask, unsigned
-    tmpCtx.signingContext.accountNumber = readUint32BE(workBuffer);
+    accountNumber = readUint32BE(workBuffer);
     workBuffer += 4;
     dataLength -= 4;
```
```suggestion
    workBufferLength -= 4;
```

CedrikNikita replied:
Forgot to rename, after parameter was renamed

CedrikNikita commented on src/utils.h:
```diff
@@ -17,7 +17,7 @@ void getAeAddressStringFromKey(cx_ecfp_public_key_t *publicKey, char *address);
 uint32_t readUint32BE(uint8_t *buffer);
 
-void sign(uint8_t *data, uint32_t dataLength, uint8_t *signature);
+void sign(uint32_t accountNumber, uint8_t *data, uint32_t dataLength, uint8_t *signature);
```
```suggestion
void sign(uint32_t accountNumber, uint8_t *data, uint16_t dataLength, uint8_t *signature);
```

CedrikNikita replied:
So the declaration and definition would match

davidyuk replied:
I don't agree, I think `dataLength` should have the same size as `hash_len` of `cx_eddsa_sign` function because actually, it is a wrapper around it.

agrojean-ledger committed:
Update CI

agrojean-ledger committed:
Merge pull request #31 from LedgerHQ/update-app

Update app : features, UI, tests, CI.

agrojean-ledger committed:
Use json + serde for Tx deserialization.

agrojean-ledger committed:
Small improvements in APDU handling

agrojean-ledger committed:
Code cleanup

kingofpayne committed:
Destination address now stored as u8 array

kingofpayne committed:
Use hex crate for hexadecimal formatting

kingofpayne committed:
Amount now as coin string and u64

kingofpayne committed:
Fixed error management in ui_display_tx

kingofpayne committed:
Minor code improvement

Better use of rust expression pattern matching.

agrojean-ledger committed:
Update dependencies to latest SDK versions for update .elf metadatas

agrojean-ledger committed:
Update CI with reusable workflows

agrojean-ledger committed:
Update app manifest to add pytest directory field.

agrojean-ledger committed:
Ignore word "crate" in misspelling checks.

agrojean-ledger committed:
Rename python tests checks workflow for clarity.

agrojean-ledger committed:
Add "name" field in Cargo.toml [package.metadata.ledger] category

agrojean-ledger committed:
Fix static analysis

agrojean-ledger committed:
Update pending review function

agrojean-ledger committed:
Add "clear pending review" fixture in python tests.

agrojean-ledger committed:
Use v1 of reusable workflows in CI as the work branch for rust support has been merged.

kingofpayne committed:
Destination address to uppercase

agrojean-ledger committed:
Merge pull request #35 from LedgerHQ/update-ci

Update CI

agrojean-ledger committed:
Fix static analysis

agrojean-ledger committed:
Fix python tests lint and type checking

agrojean-ledger committed:
Update tests snapshots

agrojean-ledger committed:
Merge pull request #34 from LedgerHQ/use-serde

Use serde

kingofpayne committed:
Improved APDU parsing logic

kingofpayne committed:
Removed unnecessary data length checks

kingofpayne committed:
Code simplification

sgliner-ledger committed:
[auto]: add PR template

yogh333 committed:
Feat/use ledger device sdk only (#37)

* ui gadgets are provided by device_sdk now

* update dependencies

* Bumped version 1.1.0

* Ragger tests OK \o/

* dependencies pulled from crates.io

kingofpayne committed:
Fixed imports due to SDK changes

kingofpayne committed:
Minor refactoring on CLA checking

kingofpayne committed:
Cargo fmt

kingofpayne committed:
Merge branch 'main' into better-commands-parsing

kingofpayne committed:
Removed unused import

kingofpayne committed:
Fixed compilation error

kingofpayne committed:
Minor fixes following code review

kingofpayne committed:
Removed status word duplicate

kingofpayne committed:
Merge pull request #39 from LedgerHQ/better-commands-parsing

Better commands parsing

agrojean-ledger committed:
Update README

Commit:
main.rs: Use display_pending_review() from SDK

xchapron-ledger committed:
Merge pull request #41 from LedgerHQ/xch/display_pending_review

main.rs: Use display_pending_review() from SDK

agrojean-ledger committed:
Merge pull request #40 from LedgerHQ/update-readme

Update README

agrojean-ledger committed:
Fix README typos.

agrojean-ledger committed:
Merge pull request #42 from LedgerHQ/fix-readme-typos

Fix README typos.

kingofpayne committed:
Use Comm::set_expected_cla from the new SDK (#44)

Use Comm::set_expected_cla from the new SDK

kingofpayne committed:
Bug fix and improvements in BIP32 path parsing (#45)

* Bug fix and improvements in BIP32 path parsing

Fixes a minor bug in size checking during BIP32 path parsing, and
created a new Bip32Path struct to store a path and its length in a more
rusty way.

yhql committed:
remove testing messages (#48)

agrojean-ledger committed:
Add Stax support

yogh333 committed:
Add pending_review_screen as an app's feature only

agrojean-ledger committed:
Test/multifield review refactor (#50)

* Use last SDK crates

* bump minor version

* Update snapshots and test_version after version bump

---------

Co-authored-by: GroM <jerome.caporossi@gmail.com>
Co-authored-by: GroM <jerome.caporossi@ledger.fr>

yogh333 committed:
Bump version and update tests accordingly

yogh333 committed:
Merge pull request #51 from LedgerHQ/jca/pending_review_feature

Add pending_review_screen as an app's feature only

yogh333 committed:
Bump SDK version

yogh333 committed:
Merge pull request #52 from LedgerHQ/jca/bump_sdk_version

Bump SDK version

sgliner-ledger committed:
[auto] Update screenshot

agrojean-ledger committed:
Small updates for NBGL sync integration.

yogh333 committed:
Use last Rust SDK, not needed anymore to use sys crate and unsafe calls

yogh333 committed:
Merge pull request #53 from LedgerHQ/jca/update_sdk_version

Use last Rust SDK, not needed anymore to use sys crate and unsafe calls

yogh333 committed:
Update Readme: Ragger tests can be launched without duplicating buildtrees

brianp committed:
Update nightly version to match Docker builds

brianp committed:
swap command arguments to correct order

bboilot-ledger committed:
Update rust SDK and fix keccak256 declaration

agrojean-ledger committed:
Use new include_gif macro to generate NBGL glyphs.

bboilot-ledger committed:
Update nanox and nanos golden snapshots

bboilot-ledger committed:
Merge pull request #57 from LedgerHQ/bboilot/update_sdk

Update rust SDK and fix keccak256 declaration

bboilot-ledger committed:
Move rustup default next to rustup install

bboilot-ledger committed:
Merge pull request #56 from brianp/readme-fixes

Readme fixes

yogh333 committed:
Merge pull request #54 from LedgerHQ/yogh333/update_readme

Update Readme: Ragger tests can be launched without duplicating build…

agrojean-ledger committed:
Update code + test snapshots following C string conversion support in Rust SDK.

agrojean-ledger committed:
Add Flex support.

agrojean-ledger committed:
Add settings example.

kingofpayne committed:
Use of heap feature for simpler code

agrojean-ledger committed:
Add dashboard icons for Stax and Flex.

agrojean-ledger committed:
Update tests snapshots following version bump in Cargo.toml

agrojean-ledger committed:
Use init_comm from nbgl rust module to initialize Comm object reference
used by nbgl api calls.

agrojean-ledger committed:
Update test_version test to parse version directly from Cargo.toml

agrojean-ledger committed:
Update get pub key tests snapshots to show status screens.

agrojean-ledger committed:
Update cargo.lock

agrojean-ledger committed:
Fix TOML parsing library import.

agrojean-ledger committed:
Fix static analysis.

agrojean-ledger committed:
Fix icon files bit depth and update tests snapshots.

agrojean-ledger committed:
Update README

agrojean-ledger committed:
PR review.

agrojean-ledger committed:
Update usage.md

agrojean-ledger committed:
Merge pull request #49 from LedgerHQ/test-stax-build

Add Stax and Flex support

agrojean-ledger committed:
Update SDK version for lock-screen fix on Stax / Flex.

agrojean-ledger committed:
Merge pull request #59 from LedgerHQ/update-sdk

Update SDK version for lock-screen fix on Stax / Flex.

yogh333 committed:
Ragger tests OK for Nano S+,X, Stax and Flex

yogh333 committed:
Use last Rust SDK + Bump version

yogh333 committed:
Remove Nano S support

yogh333 committed:
Restore crab.gif

agrojean-ledger committed:
Merge pull request #58 from LedgerHQ/feat/heap

Use of heap feature for simpler code

fbeutin-ledger committed:
Update Flex snapshots

fbeutin-ledger committed:
Merge pull request #62 from LedgerHQ/fbe/update_flex_snapshots

Update Flex snapshots

yogh333 committed:
Update snapshots

yogh333 committed:
Bump version

yogh333 committed:
Update snapshots after version bump

yogh333 committed:
Merge pull request #63 from LedgerHQ/y333/heap_is_default

Update snapshots

agrojean-ledger committed:
Update rust SDK version + update tests snapshots.

agrojean-ledger committed:
Fix Cargo.lock

agrojean-ledger committed:
Fix ragger version.

agrojean-ledger committed:
Merge pull request #65 from LedgerHQ/update-sdk

Update rust SDK version + update tests snapshots.

agrojean-ledger committed:
Update rust SDK + include_gif macro versions.

agrojean-ledger committed:
Update Cargo.lock

agrojean-ledger committed:
Merge pull request #66 from LedgerHQ/update-sdk

Update rust SDK + include_gif macro versions.

agrojean-ledger committed:
Implement status screens for Stax/Flex now that they are not managed by SDK.

agrojean-ledger committed:
Merge pull request #67 from LedgerHQ/update-status-display-for-nbgl-devices

Implement status screens for Stax/Flex now that they are not managed …

yogh333 committed:
Bump SDK version and update snapshots

yogh333 committed:
Merge pull request #72 from LedgerHQ/y333/update_sdk_version

Bump SDK version and update snapshots

fbeutin-ledger committed:
Add an option for opening PRs with golden run result in CI

fbeutin-ledger committed:
Merge pull request #73 from LedgerHQ/fbe/golden_run_CI

Add an option for opening PRs with golden run result in CI

github-actions[bot] committed:
[auto-update] Update Ragger snapshots

fbeutin-ledger committed:
Merge pull request #74 from LedgerHQ/auto/update_snapshots-6adc1b1

[auto-update] Update Ragger snapshots

github-actions[bot] committed:
[auto-update] Update Ragger snapshots

yogh333 committed:
Merge pull request #75 from LedgerHQ/auto/update_snapshots-f4f483b

[auto-update] Update Ragger snapshots

yogh333 committed:
Use sync NbglHomeAndSettings and stop redraw screen for each received APDU

yogh333 committed:
Fix clippy issues

yogh333 committed:
Update function's name which show status + home screen

yogh333 committed:
Use single loop

yogh333 committed:
Fix clippy issue

yogh333 committed:
Use last SDK and bump version

yogh333 committed:
Update snapshots

yogh333 committed:
Remove useless binding

yogh333 committed:
Refactor settings management

yogh333 committed:
Merge pull request #76 from LedgerHQ/y333/use_sync_nbgl_homeandsettings

Use async NbglHomeAndSettings and stop redrawing screen for every received…

yogh333 committed:
Call get_mut when retrieving settings

yogh333 committed:
Update dependencies

yogh333 committed:
Update SDK and bump version

yogh333 committed:
Update snapshots

yogh333 committed:
Update BP

yogh333 committed:
Use last SDK + update snapshots

yogh333 committed:
Merge pull request #77 from LedgerHQ/y333/settings_management

Y333/settings management

yogh333 committed:
Create dependabot.yml

yogh333 committed:
Merge pull request #78 from LedgerHQ/yogh333-patch-1

Create dependabot.yml

yogh333 committed:
Heap size can be set at application level

dependabot[bot] committed:
Bump ledger_device_sdk from 1.17.2 to 1.17.3

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.17.2 to 1.17.3.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.17.2...ledger_device_sdk@1.17.3)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yogh333 committed:
Merge pull request #81 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.17.3

Bump ledger_device_sdk from 1.17.2 to 1.17.3

yogh333 committed:
Add explnations

yogh333 committed:
Bump Rust SDK versions

yogh333 committed:
Merge pull request #82 from LedgerHQ/y333_20241001/heap_size_mngt

Set heap size using recommended presets

yogh333 committed:
Bump SDK version

yogh333 committed:
Merge pull request #84 from LedgerHQ/y333_241021/bump_sdk_version

Bump SDK version

dependabot[bot] committed:
Bump ledger_device_sdk from 1.17.5 to 1.18.1

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.17.5 to 1.18.1.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.17.5...ledger_device_sdk@1.18.1)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yogh333 committed:
Update README

yogh333 committed:
Merge pull request #86 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.18.1

Bump ledger_device_sdk from 1.17.5 to 1.18.1

yogh333 committed:
Remove --user when running docker

yogh333 committed:
Merge pull request #87 from LedgerHQ/y333_241025/update_readme

Update README

tdejoigny-ledger committed:
Update build_and_functional_tests.yml comment

tdejoigny-ledger committed:
Merge pull request #89 from LedgerHQ/tdj/update_comment

Update build_and_functional_tests.yml comment

dependabot[bot] committed:
Bump ledger_device_sdk from 1.18.1 to 1.18.3

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.18.1 to 1.18.3.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.18.1...ledger_device_sdk@1.18.3)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yogh333 committed:
Merge pull request #93 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.18.3

Bump ledger_device_sdk from 1.18.1 to 1.18.3

dependabot[bot] committed:
Bump ledger_device_sdk from 1.18.3 to 1.18.4

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.18.3 to 1.18.4.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.18.3...ledger_device_sdk@1.18.4)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

tdejoigny-ledger committed:
Merge pull request #94 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.18.4

Bump ledger_device_sdk from 1.18.3 to 1.18.4

yogh333 committed:
Disallow references to static mut

yogh333 committed:
Update Cargo.lock

yogh333 committed:
Update Cargo config files

yogh333 committed:
Fix Clippy issue

yogh333 committed:
Bump SDK version

yogh333 committed:
Updated snapshots

yogh333 committed:
Merge pull request #99 from LedgerHQ/y333_17122024/migration_to_updated_nightly

Y333 17122024/migration to updated nightly

dependabot[bot] committed:
Bump ledger_device_sdk from 1.19.1 to 1.19.4

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.19.1 to 1.19.4.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.19.1...ledger_device_sdk@1.19.4)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yogh333 committed:
Merge pull request #102 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.19.4

Bump ledger_device_sdk from 1.19.1 to 1.19.4

yogh333 committed:
Update readme

yogh333 committed:
update and fix typo

yogh333 committed:
Merge pull request #103 from LedgerHQ/y333/update_readme

Update readme

ghallak committed:
First try at get_address

ghallak committed:
TEMPORARY: Disable warnings

ghallak committed:
Add a note about get_version

ghallak committed:
Remove get_public_key command

ghallak committed:
Remove GetAppName command

ghallak committed:
Update .gitignore

ghallak committed:
First try at sign_tx

ghallak committed:
TEMPORARY: Stop signing non-spend txns

ghallak committed:
Use default-features to fix cargo warning

ghallak committed:
Use more appropriate status words for errors in sign_tx

ghallak committed:
Formatting, removing TODOs

ghallak committed:
Remove access of unknown size vec by index

ghallak committed:
Remove the 0x00 byte from the beginning of version command

ghallak committed:
Fix the handling of get version instruction

ghallak committed:
Refactor parse_header_data

ghallak committed:
Message signing

dependabot[bot] committed:
Bump ledger_device_sdk from 1.19.4 to 1.20.2

Bumps [ledger_device_sdk](https://github.com/LedgerHQ/ledger-device-rust-sdk) from 1.19.4 to 1.20.2.
- [Commits](https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk@1.19.4...ledger_device_sdk@1.20.2)

---
updated-dependencies:
- dependency-name: ledger_device_sdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

ghallak committed:
Rename sign to sign_tx in ui

ghallak committed:
Add note about RFC 8032

ghallak committed:
Fix warnings in get_address.rs

ghallak committed:
Fix warnings in sign_msg.rs

ghallak committed:
Separate status work for failed get_address parse

ghallak committed:
Replace crab with AE logo

ghallak committed:
Add NbglHomeAndSettings to TxContext

ghallak committed:
Remove all mentions of CRAB

ghallak committed:
Show home and status after get_address

ghallak committed:
Add UI for sign_msg

yogh333 committed:
Merge pull request #110 from LedgerHQ/dependabot/cargo/ledger_device_sdk-1.20.2

Bump ledger_device_sdk from 1.19.4 to 1.20.2

yogh333 committed:
Merge pull request #106 from ghallak/cargo-warn

Use default-features to fix cargo warning

yogh333 committed:
Bump SDK version

yogh333 committed:
Merge pull request #112 from LedgerHQ/y333/bump_sdk_version

Bump SDK version

ghallak committed:
Enable and fix warnings

ghallak committed:
Go back to home screen after message sign

ghallak committed:
Show the payload in sign_tx regardless of settings

ghallak committed:
Fix the display of amounts in AE

ghallak committed:
Remove useless screens from main menu

ghallak committed:
Update package info

ghallak committed:
Remove pending_review_screen

ghallak committed:
Delete the unused settings module

ghallak committed:
Hide the payload screen if the payload is empty

ghallak committed:
Show home screen after signing or rejecting a txn

ghallak committed:
Update response unpacker

ghallak committed:
Fix test_version_cmd.py

ghallak committed:
Remove tests for unknown commands

ghallak committed:
Remove boilerplate_reponse_unpacker.py

ghallak committed:
Run rust format on get_address

ghallak committed:
Fix test_app_mainmenu for all devices

ghallak committed:
Fix the test for error codes

ghallak committed:
Update command sender

ghallak committed:
Comment failing tests

ghallak committed:
Fix base58 check implementation

ghallak committed:
Add test for get_address

ghallak committed:
Comment unknown imports

ghallak committed:
Rename boilerplate_utils to utils

ghallak committed:
Rename test_pubkey to test_get_address

ghallak committed:
Test get address on 5 random account numbers

yogh333 committed:
Build Nano* version with NBGL

yogh333 committed:
Bump version

ghallak committed:
Encode message length as var length int

ghallak committed:
Tests for get_address confirmation and rejection

ghallak committed:
Enable the function is_finished only on stax and flex

ghallak committed:
Signing of arbitrary data

ghallak committed:
Format all python files

ghallak committed:
Rename boilerplate_transaction to transaction

yogh333 committed:
Use new feature name in Rust SDK

yogh333 committed:
Use last Rust SDK

ghallak committed:
Fix UI in signing files

ghallak committed:
Fix test_app_mainmenu.py instructions

ghallak committed:
Update snapshots for all devices

ghallak committed:
Use nbgl in the sign_data ui

ghallak committed:
Fix CI: Fix misspellings

ghallak committed:
Fix CI: Run cargo fmt --all

ghallak committed:
Fix CI: Update Cargo.lock

ghallak committed:
Fix CI: Change icons to grayscale squares

ghallak committed:
Fix CI: Update snapshots after icons update

ghallak committed:
Use 16x16 icons instead of 14x14

ghallak committed:
Rename icon files

ghallak committed:
Fix CI: Fix clippy warnings

ghallak committed:
Fix CI: Run cargo fmt --all

ghallak committed:
Fix CI: Fix python files warnings

ghallak committed:
Fix CI: Upgrade ledger_device_sdk

ghallak committed:
Fix CI: Fix import in transaction.py

davidyuk committed:
Merge branch 'master' into prepare-rust-master-2
